ulid/__init__.py

```diff
@@ -22,13 +22,17 @@
 __version__ = version("python-ulid")
 
 
-class validate_type:  # noqa: N801
-    def __init__(self, *types: Any) -> None:
+T = TypeVar("T", bound=type)
+ReturnType = TypeVar("ReturnType")
+
+
+class validate_type(Generic[T]):  # noqa: N801
+    def __init__(self, *types: T) -> None:
         self.types = types
 
-    def __call__(self, func: Callable) -> Callable:
+    def __call__(self, func: Callable[..., ReturnType]) -> Callable[..., ReturnType]:
```

Comment:
comment: This is where [ParamSpec](https://docs.python.org/3/library/typing.html#typing.ParamSpec) would come in handy, but it's sadly only available from Python 3.10 onwards.

Reply:
I had ParamSpec in there initially, but since we're not modifying the args/kwargs in the decorator, would we even need that explicitly defined?